rs/cli/src/cordoned_feature_fetcher.rs

```diff
@@ -0,0 +1,174 @@
+use std::{path::PathBuf, time::Duration};
+
+use decentralization::network::NodeFeaturePair;
+use futures::future::BoxFuture;
+use ic_management_types::NodeFeature;
+use itertools::Itertools;
+use log::warn;
+use mockall::automock;
+use reqwest::{Client, ClientBuilder};
+use strum::VariantNames;
+
+#[automock]
+pub trait CordonedFeatureFetcher: Sync + Send {
+    fn fetch(&self) -> BoxFuture<'_, anyhow::Result<Vec<NodeFeaturePair>>>;
+}
+
+pub struct CordonedFeatureFetcherImpl {
+    client: Client,
+    fallback_file: Option<PathBuf>,
+    offline: bool,
+}
+
+const CORDONED_FEATURES_FILE_URL: &str = "https://raw.githubusercontent.com/dfinity/dre/refs/heads/main/cordoned_features.yaml";
+
+impl CordonedFeatureFetcherImpl {
+    pub fn new(offline: bool, fallback_file: Option<PathBuf>) -> anyhow::Result<Self> {
+        let client = ClientBuilder::new().timeout(Duration::from_secs(10)).build()?;
+
+        Ok(Self {
+            client,
+            fallback_file,
+            offline,
+        })
+    }
+
+    async fn fetch_from_git(&self) -> anyhow::Result<Vec<NodeFeaturePair>> {
```

Comment:
Same as below - doesn't need self if `parse` doesn't.

Reply:
It does need it because of the client config.